OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -106,9 +120,13 @@ public void ResolveOrder(Actor self, Order order)
 	}
 
 	[Desc("Attach this to a unit to update observer stats.")]
-	public class UpdatesPlayerStatisticsInfo : TraitInfo<UpdatesPlayerStatistics> { }
+	public class UpdatesPlayerStatisticsInfo : TraitInfo<UpdatesPlayerStatistics>
+	{
+		[Desc("Count as army value in statistics")]
+		public bool CountAsArmy = false;
+	}
 
-	public class UpdatesPlayerStatistics : INotifyKilled
+	public class UpdatesPlayerStatistics : INotifyKilled, INotifyCreated, INotifyOwnerChanged
```

Comment:
I expect you will also need to implement `INotifyActorDisposing` so that you can clean up when actors are removed without being explicitly killed.

Reply:
You may need to track a flag inside this trait to keep track of whether it is currently counted towards the value, to avoid double (un)counting.

Reply:
What am actually I supposed to do in `INotifyActorDisposing`? How can I trigger it?

Reply:
I've been trying to do that but failed. I'm getting "Exception of type `System.InvalidOperationException`: Attempted to get trait from destroyed object (player 4 (not in world))". I'm not familiar with the language and engine so I'm probably missing something but it seems that I'm getting the notification too late when the context is not in a valid state to poke into the trait anymore.